staging/src/k8s.io/client-go/tools/events/interfaces.go

```diff
@@ -45,6 +45,11 @@ type EventBroadcaster interface {
 	// NewRecorder returns an EventRecorder that can be used to send events to this EventBroadcaster
 	// with the event source set to the given event source.
 	NewRecorder(scheme *runtime.Scheme, reportingController string) EventRecorder
+
+	// StartEventWatcher enables you to watch for emitted events without usage
+	// of StartRecordingToSink. This lets you also process events in a custom way
+	// NOTE: events received on your eventHandler should be copied before being used
+	StartEventWatcher(eventHandler func(event runtime.Object)) func()
```

Comment:
This is needed for testing a scheduling cycle, refactoring the tests needs more attention as it uses concurrency that isn't easy to reason about.
Also I remember why I made this part of the interface, some of the users have custom processing of events (this avoids for them write a tool that integrates with etcd + whatever custom logic).
I added a note to say that event should be copied before processing